lib/utils.js

```diff
@@ -831,3 +831,46 @@ exports.defineConstants = function(obj) {
   }
   return Object.freeze(exports.createMap(obj));
 };
+
+/**
+ * Whether current version of Node support ES modules
+ *
+ * @description
+ * Versions prior to 10 did not support ES Modules, and version 10 has an old incompatibile version of ESM.
+ * This function returns whether Node.JS has ES Module supports that is compatible with Mocha's needs,
+ * which is version 12 and older
+ *
+ * @param {Boolean} unflagged whether the support is unflagged (`true`) or only using the `--experimental-modules` flag (`false`)
+ * @returns {Boolean} whether the current version of Node.JS supports ES Modules in a way that is compatbile with Mocha
+ */
+exports.supportsEsModules = function(unflagged) {
```

Comment:
This function is called only once (?), without parameter => `unflagged` === false.
So `unflagged` doesn't make any sense.

Reply:
It does, because it was used in the tests (`esm.spec.js`). But in the test and in the change you did to the function, you assumed that any v13 is unflagged, which today is a close enough approximation to the truth...

Reply:
In the productive code `unflagged` has never been used. The method did not check wether user set the `--experimental-modules` or not. We should not adapt our productive code only to make pass our tests.

> you assumed that any v13 is unflagged

Yes, this is true for our CI tests.